pyk/src/pyk/kast/manip.py

```diff
@@ -756,6 +756,7 @@ def build_rule(
     final_constraints: Iterable[KInner] = (),
     priority: int | None = None,
     keep_vars: Iterable[str] = (),
+    definition: KDefinition | None = None,
```

Comment:
Maybe `defunc_with`? This way, the role of the `KDefinition` is expressed.

Reply:
Done.